### DISCUSSION TITLE
`docker-compose up airflow-init` hangs and never exits

### DISCUSSION THREAD

**Discussion by maln0ir** [apache/airflow#17056] [Q&A]
Hi, I'm trying to setup airflow using docker-compose like described in [official docs](https://airflow.apache.org/docs/apache-airflow/stable/start/docker.html), but got stuck on weird issue and don't know how to debug this problem.

Running `docker-compose up airflow-init` prints following:

    ~/s/airflow $ docker-compose up airflow-init
    airflow_redis_1 is up-to-date
    airflow_postgres_1 is up-to-date
    Starting airflow_airflow-init_1 ... done
    Attaching to airflow_airflow-init_1
    airflow-init_1       | BACKEND=postgresql+psycopg2
    airflow-init_1       | DB_HOST=postgres
    airflow-init_1       | DB_PORT=5432

and then just hangs.  According to docs it should exit at some point.

It looks like containers can't communicate with each other, for example:

    ~/s/airflow $ docker exec -i -t airflow_airflow-init_1 curl postgres:5432
    curl: (7) Failed to connect to postgres port 5432: Connection timed out
    ~/s/airflow $ docker exec -i -t airflow_airflow-init_1 curl redis:6379
    curl: (7) Failed to connect to redis port 6379: Connection timed out

Redis is exposed to the host, so I can check if it really works:

    ~ $ echo "PING" | ncat 127.0.0.1 6379
    +PONG

Tried fiddling with various options in compose file, but without luck.

---

System info:
- OS: Arch Linux
- Docker version 20.10.7, build f0df35096d
- docker-compose version 1.29.2

**Comment by potiuk** [comment]
Very interesting - the question was also posted here https://stackoverflow.com/questions/68415456/docker-compose-up-airflow-init-hangs-no-network-connection-between-containers/68420226#68420226 - see discussion there. It looks like weird DNS-related issue.

**Comment by maln0ir** [comment]
Nevermind, **I just misconfigured my firewall** and packets between containers were silently dropped :facepalm:

**Reply by potiuk** [reply]
Well that would be difficult to debug 👍
